src/Common/Core/Impl/Extensions/TaskExtensions.cs

```diff
@@ -56,14 +58,41 @@ public static class TaskExtensions {
         }
 
         /// <summary>
-        /// Suppresses warnings about unawaited tasks and ensures that unhandled
-        /// errors will cause the process to terminate.
+        /// Rethrows task exceptions back to the callers synchronization context if it is possible
         /// </summary>
         /// <remarks>
         /// <see cref="OperationCanceledException"/> is always ignored.
         /// </remarks>
-        public static async void DoNotWait(this Task task) {
-            await task.SilenceException<OperationCanceledException>();
+        public static void DoNotWait(this Task task) {
```

Comment:
Original implementation
```{csharp}
public static async void DoNotWait(this Task task) {
    await task
}
```
and the new one have several differences (both insert exceptions into another thread):
- Compiler independent implementation. `async`/`await` generates different code for Async CTP, 4.5 and 4.6.2. New implementation doesn't rely on this implementation.
- Performance. No async state machine, execution context capture and other side effects that are required for the general `async void` case.
- Selected `SynchronizationContext`. `AsyncVoidMethodBuilder` always uses `SynchronizationContextNoFlow`. New implementation doesn't care if context was captured, but rather checks if context isn't default.
- Continuation. Original implementation uses `MoveNextRunner.Run` which calls `AsyncMethodBuilderCore.ThrowAsync`. The difference is that when `DoNotWait` is called from background thread, exception is scheduled into `ThreadPool`. It crashes VS, but doesn't give us any idea about the issue. New implementation will not crash VS but will show the Trace.Fail dialog